pkg/kubelet/kubelet_pods.go

```diff
@@ -437,7 +441,7 @@ func ensureHostsFile(fileName string, hostIPs []string, hostName, hostDomainName
 		}
 	} else {
 		// if Pod is not using host network, create a managed hosts file with Pod IP and other information.
-		hostsFileContent = managedHostsFileContent(hostIPs, hostName, hostDomainName, hostAliases)
+		hostsFileContent = managedHostsFileContent(hostIPs, hostName, hostDomainName, hostAliases, isSupportIPv6())
```

Comment:
do we foresee a problem calling this function for each pod? or should we cache this value at startup?

@bart0sh @danwinship

Reply:
I personally don't like to have a dynamic behavior on the kubelet based on a sysctl , as some pods may get the ipv6 entries if the value changes at runtime
cc: @thockin

Reply:
If the user disables global ipv6 without restarting the kubelet, this will lead to error territory.

Reply:
@aojea I'm still not sure I fully understood why we can't [the solution you've proposed](https://github.com/kubernetes/kubernetes/pull/124758#discussion_r1594302638). It looks correct to me to check if at least one IPv6 exists in the list of pod IPs to make a decision.
However, if that method is incorrect, I agree that sysctl would be better to call on a Kubelet start.

Reply:
is not backward compatible  as @danwinship says ... fiddling with ips and dns is always scaring

Reply:
> So unless we have a good reason, simpler (read it as needed) seems better?

simpler to read is not simpler to troubleshoot :)

having a cluster that some of the pods have weird connectivity issues and other no to try to find out that some of them have missing host entries will be impossible .... in the super edge case that the missing host entries have some effect ... I prefer consistent behaviors ...  specially for something like `net/ipv6/conf/all/disable_ipv6` that is a very serious operation

Reply:
But it's not consistent, is it?  Restarting kubelet doesn't restart all pods, so any running pods will have the "old" config, which could be different from "new" config.  There's still a "moment in time" across which the behavior changes for new pods.

Reply:
fair, it just give you a point in time since the behavior has changed that is the kubelet restart and is traceable , the sysctl change is not ... but is not a very strong argument because this seems still very niche so I'm fine with any

Reply:
Yeah, I can go either way on it.  Either we read it once store the value in the Kubelet object, and pass it around as needed, or we pass around the sysctl interface and read this on the fly.  Either way is plumbing.

I don't think traceability is good in either case.  We should probably do this the same way we do resolv.conf, since both are "use some host info to configure a pod".  IIRC resolv.conf is read for each pod, not read once and cached.

Reply:
> . We should probably do this the same way we do resolv.conf, since both are "use some host info to configure a pod". IIRC resolv.conf is read for each pod, not read once and cached.

that seems a good idea, +1